bindings/java/src/org/sleuthkit/datamodel/CaseDatabaseFactory.java

```diff
@@ -419,6 +422,59 @@ private void createAccountTables(Statement stmt) throws SQLException {
+				+ "login_name TEXT, "	// login name, if available, may be null
+				+ "full_name TEXT DEFAULT NULL, "	// full name, if available, may be null
+				+ "realm_id " + dbQueryHelper.getBigIntType() + ", "		// row id for the realm, may be null if only SID is known 
+				+ "unique_id TEXT, "	// SID/UID, if available
+				+ "signature TEXT NOT NULL, "	// This exists only to prevent duplicates.  It is either �realm_id/unique_id� if unique_id is defined or realm_id/login_name� if unique_id is not defined.
+				+ "status INTEGER, "    // enabled/disabled/deleted
+				+ "admin INTEGER DEFAULT 0," // is admin account
+				+ "type INTEGER, "	// service/interactive
+				+ "created_date " + dbQueryHelper.getBigIntType() + ", "		
+				+ "UNIQUE(signature), "
+				+ "FOREIGN KEY(os_account_obj_id) REFERENCES tsk_objects(obj_id) ON DELETE CASCADE, "
+				+ "FOREIGN KEY(realm_id) REFERENCES tsk_os_account_realms(id) )");
+		
+		stmt.execute("CREATE TABLE tsk_os_account_attributes (id " + dbQueryHelper.getPrimaryKey() + " PRIMARY KEY, "
+				+ "os_account_obj_id " + dbQueryHelper.getBigIntType() + " NOT NULL, "
+				+ "host_id " + dbQueryHelper.getBigIntType() + ", " 
+				+ "source_obj_id " + dbQueryHelper.getBigIntType() + ", " 	
+				+ "attribute_type_id " + dbQueryHelper.getBigIntType() + " NOT NULL, "
+				+ "value_type INTEGER NOT NULL, "
+				+ "value_byte " + dbQueryHelper.getBlobType() + ", "
+				+ "value_text TEXT, "
+				+ "value_int32 INTEGER, value_int64 " + dbQueryHelper.getBigIntType() + ", "
+				+ "value_double NUMERIC(20, 10), "
+				+ "FOREIGN KEY(os_account_obj_id) REFERENCES tsk_os_accounts(os_account_obj_id) ON DELETE CASCADE, "
+				+ "FOREIGN KEY(host_id) REFERENCES tsk_hosts(id), "
+				+ "FOREIGN KEY(source_obj_id) REFERENCES tsk_objects(obj_id), "		
+				+ "FOREIGN KEY(attribute_type_id) REFERENCES blackboard_attribute_types(attribute_type_id))");	
+		
+		stmt.execute("CREATE TABLE tsk_os_account_instances (id " + dbQueryHelper.getPrimaryKey() + " PRIMARY KEY, "
+				+ "os_account_obj_id " + dbQueryHelper.getBigIntType() + " NOT NULL, "
+				+ "data_source_obj_id " + dbQueryHelper.getBigIntType() + " NOT NULL, " 
+				+ "host_id " + dbQueryHelper.getBigIntType() + " NOT NULL, "
+				+ "instance_type INTEGER NOT NULL, "	// PerformedActionOn/ReferencedOn
+				+ "UNIQUE(os_account_obj_id, data_source_obj_id, host_id), "
+				+ "FOREIGN KEY(os_account_obj_id) REFERENCES tsk_os_accounts(os_account_obj_id) ON DELETE CASCADE, "
+				+ "FOREIGN KEY(data_source_obj_id) REFERENCES tsk_objects(obj_id), "
+				+ "FOREIGN KEY(host_id) REFERENCES tsk_hosts(id))");
+		
+		stmt.execute("CREATE TABLE tsk_data_artifact ( "
+				+ "artifact_obj_id " + dbQueryHelper.getBigIntType() + " NOT NULL, "
+				+ "os_account_obj_id " + dbQueryHelper.getBigIntType() + " NOT NULL, "
+				+ "FOREIGN KEY(artifact_obj_id) REFERENCES blackboard_artifacts(artifact_obj_id) ON DELETE CASCADE, "
```

Comment:
I think that it is legal for artifact_obj_id to be both a PK and an FK and a table should have a PK to be 1NF.

Reply:
done.